examples/experimental/onnx/mobilenet_v2/main.py

```diff
@@ -0,0 +1,168 @@
+
+    # Step 5: Apply quantization algorithm.
+
+    # Define the transformation method. This method should take a data item returned
+    # per iteration through the `data_source` object and transform it into the model's
+    # expected input that can be used for the model inference.
+    input_name = onnx_model.graph.input[0].name
+    def transform_fn(data_item):
+        images, _ = data_item
+        return {input_name: images.numpy()}
+
+    # Wrap framework-specific data source into the `nncf.Dataset` object.
+    calibration_dataset = nncf.Dataset(data_source, transform_fn)
+
+    # Quantization of the ONNX model. The `quantize` method expects an ONNX
+    # model as input and returns an ONNX model that has been quantized using
+    # the calibration dataset.
+    quantized_model = nncf.quantize(onnx_model, calibration_dataset)
+
+    # Step 6: Save the quantized model.
+    quantized_model_path = MODEL_DIR / 'quantized_mobilenet_v2.onnx'
+    onnx.save(quantized_model, quantized_model_path)
+    print(f"The quantized model has been saved in: {quantized_model_path}")
+
+    # Step 7: Compare the accuracy of the original and quantized models.
+    print('Checking the accuracy of the original model:')
+    result = validate(onnx_model,
+                      data_source,
+                      providers=['OpenVINOExecutionProvider'],
+                      provider_options=[{'device_type' : 'CPU_FP32'}])
+    print(f'The original model accuracy@top1: {result:.3f}')
+
+    print('Checking the accuracy of the quantized model:')
+    quantized_result = validate(quantized_model,
+                                data_source,
+                                providers=['OpenVINOExecutionProvider'],
+                                provider_options=[{'device_type' : 'CPU_FP32'}])
+    print(f'The quantized model accuracy@top1: {quantized_result:.3f}')
+
+
+def create_data_source() -> torch.utils.data.DataLoader:
+    """
+    Creates validation ImageNet dataset.
+
+    :return: The instance of the `torch.utils.data.DataLoader`.
+    """
+    val_dir = DATASET_DIR / 'val'
+    # Transformations were taken from [here](https://pytorch.org/hub/pytorch_vision_mobilenet_v2).
+    preprocess = torchvision.transforms.Compose([
+        torchvision.transforms.Resize(256),
+        torchvision.transforms.CenterCrop(224),
+        torchvision.transforms.ToTensor(),
+        torchvision.transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]),
+    ])
+    val_dataset = torchvision.datasets.ImageFolder(val_dir, preprocess)
+    val_dataloader = torch.utils.data.DataLoader(val_dataset,
+                                                 batch_size=1,
+                                                 shuffle=False)
+    return val_dataloader
+
+
+def validate(model: onnx.ModelProto,
+             val_loader: torch.utils.data.DataLoader,
+             providers: List[str],
+             provider_options: List[Dict[str, str]],
```

Comment:
Should these parameters really be here? Is there a case when we need to change them?

Reply:
Good question. I did this to explicitly show the user that we are using `OpenVINOExecutionProvider` in the `run_example` function https://github.com/openvinotoolkit/nncf/blob/a6ebb72aa6ed629c17bb908fe0fef43f6546f466/examples/experimental/onnx/mobilenet_v2/main.py#L92